### DISCUSSION TITLE
[Feature request] Allow UNIX socket support for InfluxDB

### DISCUSSION THREAD

**Discussion by simonvanderveldt** [grafana/grafana#36992] [Data Sources]
InfluxDB supports exposing its HTTP service over a UNIX domain socket, see https://docs.influxdata.com/influxdb/v1.3/administration/config/#unix-socket-enabled-false

It would be nice if grafana would support this. It would allow running Grafana against a local InfluxDB server without opening up InfluxDB to the outside.
Also it would allow Grafana to be used in a container without opening InfluxDB to the outside. And, even better, Grafana wouldn't need to know the IP address of the host it's running in because the socket can be mounted as a volume.

**Comment by akshaychhajed** [comment]
Seems quite useful feature. I will try to submit PR for this.
Thanks @simonvanderveldt

**Comment by s0xDk** [comment]
Any update on this?

**Comment by akshaychhajed** [comment]
let me take another look and post my findings.

**Comment by Majkl578** [comment]
Any update @akshaychhajed?

[According to Go docs](https://golang.org/pkg/net/#Dial) `net` supports Unix sockets out of box so it should only be matter of updating settings validation to allow unix:// and properly configuring Dial().

```
Dial("unix", "/run/influxdb/influxdb.sock")
```

**Comment by SamuelToh** [comment]
>  It would allow Grafana to be used in a container without opening InfluxDB to the outside.

Without this feature I think this is still possible.
I believe you can accomplish easiest by using a `docker-compose` file. 

Create a `influxdb` and `grafana` service definitions. For the `influxdb` entry use the `expose` attribute to expose the api server's port (e.g. 8086) to the other services (in this case `grafana`). 

Then in your `Grafana`'s data source the URL would be `http://influxdb:8086` where the word `influxdb` equates to the name of your influxdb service defined in the `docker-compose` file.

**Comment by akshaychhajed** [comment]
@torkelo @bergquist no one is assigned to this issue/feature. Can one of you review or assign reviewer? (PR #15747 )

thanks @SamuelToh . That would work too. 
This feature would not be limited to influxdb though.
It can support any datasource serving http over unix-socket.
It would provide more flexibility.

**Comment by Sherulez** [comment]
Looking forward to this feature 👍

**Comment by gabor** [comment]
the current version of influxdb is 2.0.something, and there does not seem to be an option to use unix sockets, or at least i could not find it.  is anyone using unix sockets with influx2?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82953. Feel free to continue the discussion around this there. Thank you!
